students/timm67/session05/mailroom_part4.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env/python3
+
+"""
+Tim Meese
+Au2018-Py210B
+Mailroom Part 4 assignment
+"""
+
+import collections
+from collections import defaultdict
+
+# Revised donor database uses tuples for names instead of a single string
+
+donors = {
+     ('Jim', 'Tillson'): [5.00, 20.00],
+     ('Barb', 'Langley'): [10.00, 20.00, 100.00],
+     ('Jen', 'Garfield'): [10.00, 20.00, 5.00],
+     ('Rex', 'Miller'): [20.00, 20.00, 5.00],
+     ('Tony', 'Blake'): [20.00, 20.00, 10.00]
+     }
+
+exitMail = False
+
+
+def prompt_donor():
+    global donors
+    donor_found = False
+    donor_fname = input("Enter donor first name ('list' for all donors) : ")
+    if donor_fname == 'list':
+        for name in donors.keys():
+            print("{}, {} ".format(name[0], name[1]))
+        donor_fname = input("Enter donor first name : ")
+    donor_lname = input("Enter donor last name : ")
+    while True:
+        donor_amt = input("Enter donation value: ")
+        try:
+            donor_amt = float(donor_amt)
+            break
+        except ValueError:
+            print("Please enter a numeric donation value")
+
+    # Not optimal for optimization via list comprehension
+    for donor in donors.keys():
+        if donor[0] == donor_fname and donor[1] == donor_lname:
```

Comment:
Your `donors` is a dict, but you are doing a sequential search (like on a list), which is not desired. I see that your keys are tuples, so you could have done something like the following:
```
if (donor_fname, donor_lname) in donors:
    donors[(donor_fname, donor_lname)].append(donor_amt)
else:
    donors[(donor_fname, donor_lname)] = [donor_amt]
```
Better than this is to use `defaultdict(list)` for `donors`, and just avoid the `else` part all together (no need to check `if` condition if `donors` is a `defaultdict`).